src/core_landice/mode_forward/mpas_li_iceshelf_melt.F

```diff
@@ -0,0 +1,690 @@
+
+         endif   ! ice is present
+      enddo   ! iCell
+
+      ! change units from m/yr to kg/m2/s
+      floatingBasalMassBal(:) = floatingBasalMassBal(:) * config_ice_density / scyr
+
+
+    end subroutine basal_melt_thwaites_seroussi
+!-----------------------------------------------------------------------
+
+
+!|||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||
+!
+!  !  routine iceshelf_melt_param_from_temp_profile
+!
+!> \brief Calculate ice shelf melt rate using parameterization based on ocean temperature profile
+!> \author Matthew Hoffman, based on parameterization developed by Xylar Asay-Davis
+!> \date   October 2018
+!> \details  Melt rate parameterization by Xylar Asay-Davis used in Thwaites variability paper
+!> in prep. by Hoffman et al.
+!> Parameterization takes a ocean temperature profile and sill depth as input.
+!> The temp. profile is adjusted for cavity temperature behind the sill,
+!> assuming constant temp. below the sill depth.
+!> Then a local temp profile in the boundary layer is estimated assuming exponential
+!> decay from the cavity profile to a plume profile at infinite distance downstream
+!> from the grounding line, accounting for the pressure-melt temperature.
+!> The melt rate is calculated as proportional to the  product of the boundary layer
+!> temperature profile and the mean cavity temperature.
+!-----------------------------------------------------------------------
+
+    subroutine iceshelf_melt_param_from_temp_profile(nCellsSolve, lowerSurface, cellMask, &
+                    config_sea_level, config_ice_density, print_debug, floatingBasalMassBal, err)
+
+      !-----------------------------------------------------------------
+      ! input variables
+      !-----------------------------------------------------------------
+      integer, intent(in) :: nCellsSolve !< number of locally owned cells
+      real (kind=RKIND), dimension(:), intent(in) :: lowerSurface !< ice shelf draft
+      integer, dimension(:), intent(in) :: cellMask !< bitmask on cells
+      real (kind=RKIND), intent(in) :: config_sea_level, config_ice_density
+      logical, intent(in) :: print_debug
+
+      !-----------------------------------------------------------------
+      ! output variables
+      !-----------------------------------------------------------------
+      real (kind=RKIND), dimension(:), intent(out) :: floatingBasalMassBal !< BMB for ice shelves
+      integer, intent(out) :: err
+
+
+      !-----------------------------------------------------------------
+      ! local variables
+      !-----------------------------------------------------------------
+      ! Some constants - could be made into namelist parameters as needed
+      real(kind=RKIND), parameter :: slopeDraft = 1.0e-2_RKIND  !< characteristic slope of ice shelf draft
+      real(kind=RKIND), parameter :: plumeThickness = 30.0_RKIND  !< characteristic plume thickness (m)
```

Comment:
I would think `slopeDraft` and `plumeThickness` should eventually become namelist options if `kappa` is a namelist parameter.  `plumeThickness` in particular might be something to tune to observations.  I guess the point is that I wouldn't make these two constants in the same sense as `E0` and `Stanton`, which really are intended to be universal constants.